### ISSUE TITLE
Remove segment from uploaded segmentation

### ISSUE THREAD

**Issue by pedgomgal1** [scalableminds/webknossos#8435]
## Context
 I have uploaded a raw volume + its paired segmentation image (.tif files) from a 3d epithelium and cannot remove the instances (segments) from the segmentation layer by right-clicking in the segment of interest in the segment list. The segment removal option ("Delete segment's data") is disabled. I can add new segments and modify existing ones but cannot remove either the preexisting or new ones.

## Expected Behavior

I was expecting being able to remove any of the segments composing my segmentation layer.

Thanks for the support! Best

**Comment by philippotto** [comment]
Hi Pedro, thanks for opening this issue. Currently, the "Delete segment's data" feature is only enabled for volume annotations that were created within WK. The reason is that externally generated data can be quite large and deleting a segment can potentially require mutating a large volume. However, we might be able to work around this somehow. Let me ask a few questions for that:

- How large is the segmentation data (i.e., the bounding box size)?
- What exactly do you want to do? for example, one could also make these segments invisible if you are only concerned with the optics. Yet another way would be to use the floodfill tool (but that is limited in size, as well). if I understand your use case better, I might be able to help you out directly.

Looking forward to hear from you!

**Comment by pedgomgal1** [comment]
Hi @philippotto, thanks for addressing my issue. Let's try to shed a bit of light on it. My images at this point are not very large as they are confocal/lightshed microscopy images from epithelia. They are around 2048x2024x200 (as maximum). 

I understand when you mention large volumes, you are having in mind connectomic-like 
(EM) volumes. Despite I am also working in other projects using EM volumes at TB scale, this is not the case. The epithelia are ridiculously much simpler than neurons, but still complex enough to challenge our segmention CNNs.

What I would like to do is to remove directly the wrongly segmented epithelial cells ("segments") and proofread it by using your AI (SAM) tool in few shots. Sometimes these cells are over- or under- segmented and the easiest thing to fix them is to remove the segment completely and paint it again from scratch. 

My intention is to create a refined training dataset to feed a CNN for segmentation purposes. However, initially the data that I have to correct coming from a pretrained general model (Cellpose) brings usually a lot of mistakes. I was interested in using Webknossos because its ease to annotate cell instances using the AI tool. Your SAM tool works like a charm in my images, even along the Z axis. 

I'd highly appreciate your support in this matter as I am hunger to incorporate Webknossos to my image processing pipeline.

All the best,
Pedro

**Comment by philippotto** [comment]
Thanks for providing the details! I see the following options for you:

1) Use the floodfill tool to delete each segment. For that, you can set the active segment id to 0 in the status bar and set the floodfill to 3d (in the tool bar). if a segment is too large, the floodfill will terminate early and create a bounding box for the area that was covered successfully. in that case, you would need to use the floodfill repeatedly until the segment is fully deleted.
2) Mark all cells that you want to delete (e.g., with skeleton nodes) and then run a python script using our [libs](https://docs.webknossos.org/webknossos-py/index.html) to set the corresponding segment ids to 0.
3) Wait for us to implement the segment-deletion on full datasets (up to a certain size probably). However, I cannot make any promises when (and whether) this will happen, as we are quite busy with other plans at the moment.

I hope this helps somehow 🍀

**Comment by pedgomgal1** [comment]
Thanks a lot! sorted. BW
